File: platform/src/main/java/ru/abondin/hreasy/platform/service/dto/EmployeeDto.java
package ru.abondin.hreasy.platform.service.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.abondin.hreasy.platform.service.skills.dto.SkillDto;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * Employee information, available to any authenticated user
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EmployeeDto {
    private Integer id;
    private String displayName;
    private String sex;
    private SimpleDictDto department;
    private CurrentProjectDictDto currentProject;
    private SimpleDictDto position;
    private SimpleDictDto officeLocation;
    private SimpleDictDto ba;
    private String email;
    private String telegram;
    private OffsetDateTime telegramConfirmedAt;
    private boolean hasAvatar;
    private List<SkillDto> skills = new ArrayList<>();
}

File: platform/src/main/kotlin/ru/abondin/hreasy/platform/api/employee/EmployeeDto.kt
package ru.abondin.hreasy.platform.api.employee

import java.time.LocalDate


/**
 * Employee information, available to any authenticated user
 */
data class EmployeeDto(
        var id: Int?,
        var lastname: String?,
        var firstname: String?,
        var patronymicName: String?,
        var displayName: String?,
        var birthday: LocalDate?,
        var sex: String?,
        var department: SimpleDictDto?,
        var currentProject: SimpleDictDto?,
        var email: String?,
        var phone: String?,
        var skype: String?,
        var hasAvatar: Boolean = false
)

/**
 * Full representation of Employee DTO for admin APIs.
 */
data class FullEmployeeDto(
        var id: Int?,
        var lastname: String?,
        var firstname: String?,
        var patronymicName: String?,
        var birthday: LocalDate?,
        var sex: String?,
        var dateOfEmployment: LocalDate?,
        var department: SimpleDictDto?,
        var position: SimpleDictDto?,
        var level: SimpleDictDto?,
        var workType: String?,
        var workDay: String?,
        var email: String?,
        var phone: String?,
        var skype: String?,
        var registrationAddress: String?,
        var documentSeries: String?,
        var documentNumber: String?,
        var documentIssuedBy: String?,
        var documentIssuedDate: LocalDate?,
        var foreignPassport: String?,
        var cityOfResidence: String?,
        var englishLevel: String?,
        var familyStatus: String?,
        var spouseName: String?,
        var children: String?,
        var dateOfDismissal: LocalDate?
) {
}